### PR TITLE
feat(scheduling): [MC-669] Allow filtering in Schedule View

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Add filters to each day's data on the Schedule page. Wireframes in ticket.

Acceptance Criteria (copied over from JIRA):

- For a given day on the schedule, allow Filtering by Topic, Publisher, or Type 
- Provide Filtering for each day on the page separately (for example, if I am looking at the 24th & 25th on the page at once I can use the filter for the 24th or filter the 25th)
- Changing the date resets all of the filters
- Only one filter is available at a time

## Video Walkthrough

Note: PR currently deployed to Dev, so can be test-driven there: https://curation-admin-tools.getpocket.dev/

https://github.com/Pocket/curation-admin-tools/assets/22447785/310d3799-99bd-4f6d-afc5-86e71a2a8d50

## References

JIRA: https://mozilla-hub.atlassian.net/browse/MC-669

### CODE REVIEW

File: src/curated-corpus/components/ScheduleDayData/ScheduleDayData.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+/**
+ * Show all scheduled entries for a given date.
+ *
+ * @param props
+ * @constructor
+ */
+export const ScheduleDayData: React.FC<ScheduleDayDataProps> = (
+  props,
+): ReactElement => {
```
Comment (nina-py) [inline src/curated-corpus/components/ScheduleDayData/ScheduleDayData.tsx:76]: This component is a big chunk of functionality from SchedulePage, now in its own component. It's still too cumbersome to test effectively - I think it needs to be broken up further for that. The other three new components have test coverage.

File: src/_shared/components/CardActionButtonRow/CardActionButtonRow.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ interface CardActionButtonRowProps {
 }
 
 export const CardActionButtonRow: React.FC<CardActionButtonRowProps> = (
-  props
+  props,
```
Comment (nina-py) [inline src/_shared/components/CardActionButtonRow/CardActionButtonRow.tsx:39]: A whole heap of linting tweaks were needed for the app to run locally - strange, but ok.

Comment (jpetto) [reply src/_shared/components/CardActionButtonRow/CardActionButtonRow.tsx:39]: i just ran into this too! weird...

**Review by jpetto — APPROVED** [review]
this looks good to me. i tested on dev too and everything is working as expected.

i kind of expected the filters to show the selection, but if curators are good with it as-is, i don't think we need to wait. (maybe showing the selection can be a follow-up.)

**Pull request merged**